base/common/src/main/java/com/netscape/certsrv/profile/ProfileData.java

```diff
@@ -273,18 +286,339 @@ public boolean equals(Object obj) {
+            nameElement.appendChild(document.createTextNode(name));
+            pdElement.appendChild(nameElement);
+        }
+        if (description != null) {
+            Element descriptionElement = document.createElement("description");
+            descriptionElement.appendChild(document.createTextNode(description));
+            pdElement.appendChild(descriptionElement);
+        }
+        Element enabledElement = document.createElement("enabled");
+        enabledElement.appendChild(document.createTextNode(Boolean.toString(enabled)));
+        pdElement.appendChild(enabledElement);
+
+        Element visibleElement = document.createElement("visible");
+        visibleElement.appendChild(document.createTextNode(Boolean.toString(visible)));
+        pdElement.appendChild(visibleElement);
+        if (enabledBy != null) {
+            Element enabledByElement = document.createElement("enabledBy");
+            enabledByElement.appendChild(document.createTextNode(enabledBy));
+            pdElement.appendChild(enabledByElement);
+        }
+        if (authenticatorId != null) {
+            Element authenticatorIdElement = document.createElement("authenticatorId");
+            authenticatorIdElement.appendChild(document.createTextNode(authenticatorId));
+            pdElement.appendChild(authenticatorIdElement);
+        }
+        if (authzAcl != null) {
+            Element authzAclElement = document.createElement("authzAcl");
+            authzAclElement.appendChild(document.createTextNode(authzAcl));
+            pdElement.appendChild(authzAclElement);
+        }
+        Element renewalElement = document.createElement("renewal");
+        renewalElement.appendChild(document.createTextNode(Boolean.toString(renewal)));
+        pdElement.appendChild(renewalElement);
+
+        Element xmlOutputElement = document.createElement("xmlOutput");
+        xmlOutputElement.appendChild(document.createTextNode(Boolean.toString(xmlOutput)));
+        pdElement.appendChild(xmlOutputElement);
+        for (ProfileInput pi : inputs) {
+            Element piElement = pi.toDOM(document);
+            pdElement.appendChild(piElement);
+        }
+        for (ProfileOutput po : outputs) {
+            Element poElement = po.toDOM(document, "Output");
+            pdElement.appendChild(poElement);
+        }
+        if (!policySets.isEmpty()) {
+            Element policysetsElement = document.createElement("PolicySets");
+            for (Entry<String, List<ProfilePolicy>> policySet : getPolicySets().entrySet()) {
+                Element policysetElement = document.createElement("PolicySet");
+                Element idElement = document.createElement("id");
+                idElement.appendChild(document.createTextNode(policySet.getKey()));
+                policysetElement.appendChild(idElement);
+                Element valueElement = document.createElement("value");
+                for (ProfilePolicy pp : policySet.getValue()) {
```

Comment:
I think there should be one `<value>` element for each `ProfilePolicy`, so it should be in the loop.

Reply:
Agreed, fixed in next revision.